### PR TITLE
[docs-only] add app provider docs

### PR DESCRIPTION
**Pull request by wkloucek**
## Description
Document the app provider and how to use it.

## Related Issue

- Follow up https://github.com/owncloud/ocis/pull/2705

### CODE REVIEW

**Comment by wkloucek** [comment]
@pascalwengerter @kulmann could you please give this a short read?

File: docs/extensions/storage/apps.md [hunk 1]
```diff
@@ -0,0 +1,305 @@
+---
+title: "Apps"
+date: 2018-05-02T00:00:00+00:00
+weight: 10
+geekdocRepo: https://github.com/owncloud/ocis
+geekdocEditPath: edit/master/docs/extensions/storage
+geekdocFilePath: apps.md
+---
+
+oCIS is all about files. But most of the time you want to do something with files that beyond the basic upload, download and share behavior. Therefore, oCIS has a concept for apps, that can handle specific file types, so called mime types.
```
Comment (kulmann) [inline docs/extensions/storage/apps.md:10]:
```suggestion
oCIS is all about files. But most of the time you want to do something with files that is beyond the basic upload, download and share behavior. Therefore, oCIS has a concept for apps, that can handle specific file types, so called mime types.
```

File: docs/extensions/storage/apps.md [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+            }
+          ]
+        }
+      }
+    }
+  }
+}
+```
+
+Please note that there might be two or more app providers with different versions. This is not be expected to happen on a regular basis. It was designed for a possible migration period for clients when the app provider needs a breaking change.
```
Comment (kulmann) [inline docs/extensions/storage/apps.md:37]: Could you wrap this into a `hint` tag?

File: docs/extensions/storage/apps.md [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+
+## App registry
+
+The app registry is the single point where all apps register themselves and their respective supported mime types.
+
+### Mime type configuration / creation allow list
+
+The apps will register their supported mime types automatically, so that users can open supported files with them.
+
+Administrators can set default applications on a per mimetype basis and also allow the creation of new files for certain mime types. This per mime type configuration also features a description, file extension option and an icon.
```
Comment (kulmann) [inline docs/extensions/storage/apps.md:47]:
```suggestion
Administrators can set default applications on a per mime type basis and also allow the creation of new files for certain mime types. This per mime type configuration also features a description, file extension option and an icon.
```

File: docs/extensions/storage/apps.md [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+
+In order to modify the mime type config you need to set `STORAGE_APP_REGISTRY_MIMETYPES_JSON=.../mimetypes.json` to a valid JSON file with a content like this:
```
Comment (kulmann) [inline docs/extensions/storage/apps.md:49]:
```suggestion
In order to modify the mime type config you need to set `STORAGE_APP_REGISTRY_MIMETYPES_JSON=.../mimetypes.json` to a valid JSON file with content like this:
```

**Review by kulmann — CHANGES_REQUESTED** [review]
some more nitpicks

File: docs/extensions/storage/apps.md (hunk 2 above)
Comment (wkloucek) [reply docs/extensions/storage/apps.md:38]: done :+1:

**Review by kulmann — APPROVED** [review]
LGTM 📜

**Pull request merged**